### ISSUE TITLE
MethodError: no method matching degree(::Float64)

### ISSUE THREAD

**Issue by huwenboshi** [JuliaStats/StatsModels.jl#66]
I tried to create a formula like so:
@formula(T ~ 1 + 0.5*x)

But got the following error:
Warning: Number 0.5 removed from interaction term 0.5 & x
ERROR: MethodError: no method matching degree(::Float64)
Closest candidates are:
  degree(::Expr) at /u/home/s/shihuwen/.julia/v0.6/StatsModels/src/formula.jl:291
  degree(::Symbol) at /u/home/s/shihuwen/.julia/v0.6/StatsModels/src/formula.jl:289
  degree(::Integer) at /u/home/s/shihuwen/.julia/v0.6/StatsModels/src/formula.jl:288

This worked fine with version 0.0.2.

**Comment by ararslan** [comment]
I don't believe we currently support transformations in formulas, which includes scaling by a constant. I could be wrong on that though.

**Comment by sarah-ji** [comment]
I was able to get @formula(T ~ 1 + 0.5.*x) to run, is it not doing what the same thing?

**Comment by kleinschmidt** [comment]
`@formula` runs but you can't do much more than that (it can't be converted to a `ModelFrame`): 

```julia
julia> @formula(y ~ 1 + 0.5.*x)
Formula: y ~ 1 + 0.5 .* x

julia> ModelFrame(@formula(y ~ 1 + 0.5.*x), DataFrame(y=[1,2,3], x = [1,2,3]))
ERROR: MethodError: Cannot `convert` an object of type Array{Any,1} to an object of type Expr
This may have arisen from a call to the constructor Expr(...),
since type constructors fall back to convert methods.
```

**Comment by sarah-ji** [comment]
It seems that when a formula has only one element there is a problem querying it's argument. 
For example:
julia> FZ = @formula(T ~ W + Z)
Formula: T ~ W + Z

julia> FZ.rhs.args
3-element Array{Any,1}:
 :+
 :W
 :Z

julia> FF = @formula(T ~ W)
Formula: T ~ W

julia> FF.rhs
:W

julia> FF.rhs.args
ERROR: type Symbol has no field args

Any insight??

**Comment by kleinschmidt** [comment]
The RHS can be an `Expr` (as in your first case), which has an `args` field, or a `Symbol` (if there's just one RHS term).  The most natural way to access the arguments depends on what you want to do with them.  If you want to extract all the `Symbol`s from an `Expr` you'll have to process that `Expr` (which could have `Symbol` arguments as well as other `Expr`.  The source code for this package has some examples of how you can do that (e.g. https://github.com/JuliaStats/StatsModels.jl/blob/master/src/formula.jl#L309-L319)
